src/main/java/io/cryostat/core/templates/RemoteTemplateService.java

```diff
@@ -55,35 +55,42 @@ protected TemplateType providedTemplateType() {
-                                            "Document did not contain \"configuration\" element");
-                                }
-                                if (els.size() > 1) {
-                                    throw new MalformedXMLException(
-                                            "Document contains multiple \"configuration\""
-                                                    + " elements");
-                                }
-                                Element configuration = els.first();
-                                if (!configuration.hasAttr("label")) {
-                                    throw new MalformedXMLException(
-                                            "Configuration element did not have \"label\""
-                                                    + " attribute");
-                                }
-                                return configuration.attr("label").equals(templateName);
-                            })
-                    .findFirst();
+            Optional<Document> document =
+                    conn.getService().getServerTemplates().stream()
+                            .map(xmlText -> Jsoup.parse(xmlText, "", Parser.xmlParser()))
+                            .filter(
+                                    doc -> {
+                                        Elements els = doc.getElementsByTag("configuration");
+                                        if (els.isEmpty()) {
+                                            throw new MalformedXMLException(
+                                                    "Document did not contain \"configuration\""
+                                                            + " element");
+                                        }
+                                        if (els.size() > 1) {
+                                            throw new MalformedXMLException(
+                                                    "Document contains multiple \"configuration\""
+                                                            + " elements");
+                                        }
+                                        Element configuration = els.first();
+                                        if (!configuration.hasAttr("label")) {
+                                            throw new MalformedXMLException(
+                                                    "Configuration element did not have \"label\""
+                                                            + " attribute");
+                                        }
+                                        return configuration.attr("label").equals(templateName);
+                                    })
+                            .findFirst();
+            if (document.isPresent()) {
+                return Optional.of(document.get().toString());
+            } else {
+                return Optional.empty();
+            }
```

Comment:
I think handing the `Optional` explicitly like this can be avoided by just `.map()`ing the result after filtering:

```suggestion
                    return conn.getService().getServerTemplates().stream()
                            .map(xmlText -> Jsoup.parse(xmlText, "", Parser.xmlParser()))
                            .filter(
                                    doc -> {
                                        Elements els = doc.getElementsByTag("configuration");
                                        if (els.isEmpty()) {
                                            throw new MalformedXMLException(
                                                    "Document did not contain \"configuration\""
                                                            + " element");
                                        }
                                        if (els.size() > 1) {
                                            throw new MalformedXMLException(
                                                    "Document contains multiple \"configuration\""
                                                            + " elements");
                                        }
                                        Element configuration = els.first();
                                        if (!configuration.hasAttr("label")) {
                                            throw new MalformedXMLException(
                                                    "Configuration element did not have \"label\""
                                                            + " attribute");
                                        }
                                        return configuration.attr("label").equals(templateName);
                                    })
                             .map(doc -> document.toString())
                            .findFirst();
```